website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -13,9 +14,18 @@ import {
   getTasks,
   moveTask,
 } from '../../../libs/taskManager';
+import apiMessages from '../../../libs/apiMessages';
 
 let requiredGroupFields = '_id leader tasksOrder name';
 let types = Tasks.tasksTypes.map(type => `${type}s`);
+
+function canNotEditTasks (group, user, assignedUserId) {
```

Comment:
IT would be easy to abstract, just incorrect. user.isLeader, user.isManager make sense, but user.canNotEditTasks doesn't really make sense for the owning object to be the approver. 

So, you could add the function to the group and have group.userCanEditTask, but then we see that this code is duplicated as well: https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/tasks.js#L1192 . So, why are we abstracting one and not the other? 

Then, we run into a SOLID design issue on the group model (which we already have). But we can see that a better (first step) would be to create a TaskPolicy class to begin handling these permissions. Which would be out of scope of this PR.

So, sure, it is easy, but it would be support bad code that I have already brought up as an issue.